server/src/main/java/org/apache/cassandra/sidecar/utils/SidecarClientProvider.java

```diff
@@ -89,10 +104,16 @@ public void close()
         }
     }
 
+    // Exposing the used http client to be reused on periodic tasks
+    public HttpClient getHttpClient()
```

Comment:
let's not expose the http client here, instead let's just expose a method to update ssl options.
```suggestion
    public Future<Boolean> updateSSLOptions()
```